test/support.ts

```diff
@@ -0,0 +1,350 @@
+
+const distDir = path.resolve(__dirname, '../lib');
+
+chai.use(chaiDatetime);
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+
+// Using util.inspect to correctly assert objects with symbols
+// Because expect.deep.equal does not test non iterator keys such as symbols (https://github.com/chaijs/chai/issues/1054)
+chai.Assertion.addMethod('deepEqual', function deepEqual(expected, depth = 5) {
+  // eslint-disable-next-line @typescript-eslint/no-invalid-this -- this is how chai function
+  expect(inspect(this._obj, { depth })).to.deep.equal(inspect(expected, { depth }));
+});
+
+chai.config.includeStack = true;
+chai.should();
+
+// Make sure errors get thrown when testing
+process.on('uncaughtException', e => {
+  console.error('An unhandled exception occurred:');
+  throw e;
+});
+
+let onNextUnhandledRejection: ((error: unknown) => any) | null = null;
+let unhandledRejections: unknown[] | null = null;
+
+process.on('unhandledRejection', e => {
+  if (unhandledRejections) {
+    unhandledRejections.push(e);
+  }
+
+  const onNext = onNextUnhandledRejection;
+  if (onNext) {
+    onNextUnhandledRejection = null;
+    onNext(e);
+  }
+
+  if (onNext || unhandledRejections) {
+    return;
+  }
+
+  console.error('An unhandled rejection occurred:');
+  throw e;
+});
+
+afterEach(() => {
+  onNextUnhandledRejection = null;
+  unhandledRejections = null;
+});
+
+/**
+ * Returns a Promise that will reject with the next unhandled rejection that occurs
+ * during this test (instead of failing the test)
+ */
+export async function nextUnhandledRejection() {
+  return new Promise((resolve, reject) => {
+    onNextUnhandledRejection = reject;
+  });
+}
+
+/**
+   * Pushes all unhandled rejections that occur during this test onto destArray
+   * (instead of failing the test).
+   *
+   * @param destArray the array to push unhandled rejections onto.  If you omit this,
+   * one will be created and returned for you.
+   *
+   * @returns destArray
+   */
+export function captureUnhandledRejections(destArray = []) {
+  unhandledRejections = destArray;
+
+  return unhandledRejections;
+}
+
+let lastSqliteInstance: Sequelize | undefined;
+export async function prepareTransactionTest(sequelize: Sequelize) {
+  const dialect = getTestDialect();
+
+  if (dialect === 'sqlite') {
+    const p = path.join(__dirname, 'tmp', 'db.sqlite');
+    if (lastSqliteInstance) {
+      await lastSqliteInstance.close();
+    }
+
+    if (fs.existsSync(p)) {
+      fs.unlinkSync(p);
+    }
+
+    const options = { ...sequelize.options, storage: p };
+    const _sequelize = new Sequelize(sequelize.config.database, '', '', options);
+
+    await _sequelize.sync({ force: true });
+    lastSqliteInstance = _sequelize;
+
+    return _sequelize;
+  }
+
+  return sequelize;
+}
+
+export function createSequelizeInstance(options: Options = {}) {
+  options.dialect = getTestDialect();
+
+  const config = Config[options.dialect];
+
+  const sequelizeOptions = defaults(options, {
```

Comment:
Why is `odbcConnectionString` not mentioned here anymore?

Reply:
It wasn't correctly set (according to the typings, and its usage in ibmi's connection manager). The `obdcConnectionString` isn't a valid option, it's named `dialectOptions.obdcConnectionString`. Which means it was unused